### ISSUE TITLE
segfault when using --target-indexed as of 2.1.0

### ISSUE THREAD

**Issue by rzhao-2** [bbuchfink/diamond#695]
Hi, much appreciate the work that has gone into this. I attempted to run the latest version of DIAMOND with the following command:
```
~/e/diamond-2.1.6 blastx -q NC_000871.1.fa -d prefilter.dmnd -c1 --target-indexed --verbose
```
Which outputs the following:
```
diamond v2.1.6.160 (C) Max Planck Society for the Advancement of Science
Documentation, support and updates available at http://www.diamondsearch.org
Please cite: http://dx.doi.org/10.1038/s41592-021-01101-x Nature Methods (2021)

#CPU threads: 128
Scoring parameters: (Matrix=BLOSUM62 Lambda=0.267 K=0.041 Penalties=11/1)
CPU features detected: ssse3 popcnt sse4.1 avx2
Temporary directory:
#Target sequences to report alignments for: 25
Opening the database...  [0.006s]
Database: prefilter.dmnd (type: Diamond database, sequences: 49037, letters: 3775960)
Block size = 2000000000
Opening the input file...  [0s]
Opening the output file...  [0s]
Loading query sequences... Sequences = 6, letters = 74736, average length = 12456
 [0s]
Sequences = 6, letters = 74736, average length = 12456
Masking queries...  [0.011s]
Algorithm: Double-indexed
Shape configuration: 111101110111,111011010010111
Seeking in database...  [0s]
Loading reference sequences... Sequences = 49037, letters = 3775960, average length = 77
 [0.006s]
Masking reference...  [0.041s]
Initializing temporary storage...  [0.006s]
Building reference histograms...  [0.016s]
Allocating buffers... Segmentation fault (core dumped)
```
The segfault occurs in 2.1.0 as well, while 2.0.15 works fine.

Many thanks,
Rossen

**Comment by bbuchfink** [comment]
This should fix it: 82e9959fa5a811a21d03904e229776cba666f2e2

**Comment by rzhao-2** [comment]
Works now, thanks for the quick response. Looking forward to 2.1.7.
